test/ambiguities.jl

```diff
@@ -10,4 +10,8 @@ const allowable_ambiguities =
         1
     end
 
-@test length(detect_ambiguities(Base, LinearAlgebra, StaticArrays)) <= allowable_ambiguities
+@static if VERSION < v"1.6-"
```

Comment:
A minor note - this use of `@static` isn't really required (though nor is it harmful).

There's two reasons to use `@static`:
* For efficiency, if the compiler can't prove that the condition is constant (likely only relevant inside a function)
* To avoid expanding a macro (or generally letting the compiler frontend lower any code) which is inside one of the branches. (For example, imagine you wanted to use the new `@ccall` in one of the branches, but it only exists in julia-1.5.)

Reply:
You're right of course, I tend to eagerly mark as `@static` what can be because of some lowering bugs I have encountered due to the second reason you gave (when the macro isn't defined until some Julia version for instance), but it's true that in this context it is not useful.